tests/unit/test_connection.py

```diff
@@ -0,0 +1,27 @@
+from dbt.adapters.dremio.api.rest.error import DremioRequestTimeoutException
+import pytest
+from unittest.mock import patch
+from dbt.adapters.dremio.connections import DremioConnectionManager
+from requests.exceptions import HTTPError
+from dbt.exceptions import FailedToConnectException
+
+
+class TestRetryConnections:
```

Comment:
nit "TestRetryConnections" --> "TestRetryConnection" as the method name is "retry_connection"